nntrainer/include/databuffer_func.h

```diff
@@ -31,11 +31,18 @@
 #include <functional>
 #include <iostream>
 #include <memory>
+#include <nntrainer_internal.h>
 #include <thread>
 #include <vector>
 
 namespace nntrainer {
 
+/**
+ * @brief   Dataset generator callback type declaration
+ */
+typedef std::function<std::remove_pointer<ml_train_datagen_cb>::type>
```

Comment:
IMO, it'd better to:
   1. declare `ml_train_datagen_cb` -> `datagen_cb_ptr` inside here
   2. alias this ptr from capi

otherwise it lib depends on api making circular dependency.

in this case `nntrainer_internal.h <- nntrainer.h <- lib <- nntrainer_internal.h`

Reply:
My intension is to not define the same function pointer twice. I updated the dependency of `databuffer_func.h` to `nntrainer.h` than `nntrainer_internal.h`. Does that satisfy the no circular dependency?
I do not understand what you mean by aliasing?

Reply:
I might have picked wrong term. I mean... header will end up referenced both ways. (capi -> lib) and (lib <- capi)

https://github.com/nnstreamer/nntrainer/blob/f45060b9d570bdb61063790dc5ac6c1aace26646/nntrainer/meson.build#L1-L4

I think `../api/capi/include` shouldn't be included to `nntrainer_inc` if possible.
If that's the case, nntrainer-devel will be super set of capi-nntrainer-devel,
If there is `api/python` or `api/cpp` binding, it has to include `api/capi` as well and so on..

Overall making it hard to manage the package.

What I meant was this would be better if is sth like:

##### in `nntrainer.h`
```cpp
#include "databuffer_func.h"
typedef datagen_cb_ptr ml_train_datagen_cb
```

##### in `databuffer_func.h`
```cpp
// remove #include "nntrainer.h" 
typedef int (*datagen_cb_ptr)(float **input, float **label, bool *last);
```

Reply:
> #include "databuffer_func.h"
> typedef datagen_cb_ptr ml_train_datagen_cb

Using this would expose databuffer_func.h to the outside.
I will get comments from other members and update correspondingly.

Reply:
As per discussion with @myungjoo, the proposed change by this PR is fine.